les/enr_entry.go

```diff
@@ -36,7 +36,7 @@ func (e lesEntry) ENRKey() string {
 
 // setupDiscovery creates the node discovery source for the eth protocol.
 func (eth *LightEthereum) setupDiscovery(cfg *p2p.Config) (enode.Iterator, error) {
-	if cfg.NoDiscovery || len(eth.config.DiscoveryURLs) == 0 {
+	if !cfg.DiscoveryV5 || len(eth.config.DiscoveryURLs) == 0 {
```

Comment:
Is it `if !cfg.DiscoveryV5 && len(eth.config.DiscoveryURLs) == 0 {`? 
Then if DNS discovery is enabled(`len(eth.config.DiscoveryURLs) != 0`), we add the dns iterator. 
Then if the discoveryV5 is enabled(`cfg.DiscoveryV5 = true`), we add the v5 iterator?

Reply:
I added some more logging, but otherwise didn't touch any logic
```
[user@work go-ethereum]$ ./build/bin/geth --syncmode=light --nodiscover 2>&1 | grep setupDiscovery
INFO [11-26|08:19:46.548] setupDiscovery                           cfg.DiscoveryV5 =false cfg.NoDiscovery=true len(eth.config.DiscoveryURLs)=1
INFO [11-26|08:19:46.549] setupDiscovery                           DNS discovery=OFF
^C
[user@work go-ethereum]$ ./build/bin/geth --syncmode=light  2>&1 | grep setupDiscovery
INFO [11-26|08:19:52.632] setupDiscovery                           cfg.DiscoveryV5 =true cfg.NoDiscovery=true len(eth.config.DiscoveryURLs)=1
INFO [11-26|08:19:52.632] setupDiscovery                           DNS discovery=ON
^C
[user@work go-ethereum]./build/bin/geth --syncmode=light --discovery.dns="" 2>&1 | grep setupDiscovery
INFO [11-26|08:21:32.540] setupDiscovery                           cfg.DiscoveryV5 =true cfg.NoDiscovery=true len(eth.config.DiscoveryURLs)=0
INFO [11-26|08:21:32.540] setupDiscovery                           DNS discovery=OFF

```
Seems to work as expected

Reply:
Hm, actually, I don't see why it didn't work previously. Here's with the same logging, but without the changes in this PR

```
[user@work go-ethereum]$ ./build/bin/geth --syncmode=light --nodiscover 2>&1 | grep setupDiscovery
INFO [11-26|08:23:57.408] setupDiscovery                           cfg.DiscoveryV5 =false cfg.NoDiscovery=true len(eth.config.DiscoveryURLs)=1
INFO [11-26|08:23:57.408] setupDiscovery                           DNS discovery=ON
^C[user@work go-ethereum]./build/bin/geth --syncmode=light  2>&1 | grep setupDiscovery
INFO [11-26|08:24:04.923] setupDiscovery                           cfg.DiscoveryV5 =true cfg.NoDiscovery=true len(eth.config.DiscoveryURLs)=1
INFO [11-26|08:24:04.923] setupDiscovery                           DNS discovery=ON
^C
[user@work go-ethereum]$ ./build/bin/geth --syncmode=light --discovery.dns="" 2>&1 | grep setupDiscovery
INFO [11-26|08:24:37.616] setupDiscovery                           cfg.DiscoveryV5 =true cfg.NoDiscovery=true len(eth.config.DiscoveryURLs)=0
INFO [11-26|08:24:37.616] setupDiscovery                           DNS discovery=OFF
```
So previously, it errored on enabling dns discovery when it shouldn't, not the other way around

Reply:
Sorry, I was wrong about how the current master works. This is how it works before this PR: 
```
[user@work go-ethereum]$ ./build/bin/geth --syncmode=light --nodiscover 2>&1 | grep setupDiscovery
INFO [11-26|08:36:05.775] setupDiscovery                           cfg.DiscoveryV5 =false cfg.NoDiscovery=true len(eth.config.DiscoveryURLs)=1
INFO [11-26|08:36:05.775] setupDiscovery                           DNS discovery=OFF
^C
user@work go-ethereum]$ ./build/bin/geth --syncmode=light  2>&1 | grep setupDiscovery
INFO [11-26|08:36:26.362] setupDiscovery                           cfg.DiscoveryV5 =true cfg.NoDiscovery=true len(eth.config.DiscoveryURLs)=1
INFO [11-26|08:36:26.362] setupDiscovery                           DNS discovery=OFF
^C
[user@work go-ethereum]$ ./build/bin/geth --syncmode=light --discovery.dns="" 2>&1 | grep setupDiscovery
INFO [11-26|08:36:31.963] setupDiscovery                           cfg.DiscoveryV5 =true cfg.NoDiscovery=true len(eth.config.DiscoveryURLs)=0
INFO [11-26|08:36:31.964] setupDiscovery                           DNS discovery=OFF
```

Reply:
@rjl493456442 this was indeed weird and misleading. In light mode it was the DiscoveryV5 flag that was tied to --nodiscover and therefore showed whether discovery is enabled or not. Now I added a separate DnsDiscovery flag to make it cleaner.